pkg/maas/scope/cluster.go

```diff
@@ -281,3 +281,12 @@ func (s *ClusterScope) IsAPIServerOnline() (bool, error) {
 
 	return err == nil, nil
 }
+
+func (s *ClusterScope) IsCustomEndpoint() bool {
+	if infrautil.IsCustomEndpointPresent(s.MaasCluster.GetAnnotations()) {
+		s.GetDNSName()
```

Comment:
I see this is noop do we need to call this function twice

Reply:
GetDNSName does set host as DNSName, so we can keep it. If not configured it will be set.

Reply:
Let's split getter and setters we should not set with Get

Reply:
Will be handled as a new pr